### PR TITLE
[dv] Update kmac_app_monitor to remove glitch of ok_to_end

### PR DESCRIPTION
**Pull request by weicaiyang**
The update doesn't change any functionality but this will remove 0 delay
glich of ok_to_end
Signed-off-by: Weicai Yang <weicai@google.com>

### CODE REVIEW

**Review by rasmus-madsen — APPROVED** [review]
@weicaiyang ok I was not aware of the premature ending scenario
LGTM I will rebase my PR once this is merged

File: hw/dv/sv/kmac_app_agent/kmac_app_monitor.sv
Comment (weicaiyang) [reply hw/dv/sv/kmac_app_agent/kmac_app_monitor.sv]: Fixed the typo. Prefer to keep the first sentence as it's not clear to me if that is a right approach. May have a better idea once we know kmac TB more

**Pull request merged**
